streaming-binlog/src/main/java/org/openmrs/analytics/DebeziumListener.java

```diff
@@ -98,4 +100,11 @@ public void close() {
 		
 	}
 	
+	private DbeziumArgs getpassedArgs() {
+		String[] args = Runner.getArgs();
+		DbeziumArgs params = new DbeziumArgs();
+		JCommander.newBuilder().addObject(params).build().parse(args);
```

Comment:
I don't know much about JCommander but I prefer if we can find/use an argument parsing framework where for each `@Parameters` class only one instance is created and filled (by parsing command line args) and then it is accessed everywhere as a Singleton throughout the program. The way that we are doing it here it seems we are reparsing all arguments for each access to one of them. If what I suggested is not easy to do, let's improve this by only calling `getPassedArgs` only once for each `DebeziumListener` instance (that should be easy).

Reply:
fixed

Reply:
Thanks for the fix. Any thoughts on the other/bigger part of my comment? I mean is there a way to define `@Parameter` static fields in different library classes (i.e., command line args that are specific to one library only) and then do something with JCommander (or another command-line processing framework) at `main` to parse all arguments only once and fill up all those static fields with their corresponding values parsed from args?